internal/querynode/search.go

```diff
@@ -39,6 +39,7 @@ func searchOnSegments(replica ReplicaInterface, segType segmentType, searchReq *
 			defer wg.Done()
 			seg, err := replica.getSegmentByID(segID, segType)
 			if err != nil {
+				errs[i] = fmt.Errorf("searchOnSegments failed, err = %s", err)
```

Comment:
This line will has error in normal cases since getSegmentByID will query sealed segment ID with growing type as trial

Reply:
Also, missing a segment is expected in hand-off term.